packages/odyssey-react-mui/src/labs/FileUpload.tsx

```diff
@@ -0,0 +1,335 @@
+          fill="#E1E1E1"
+        />
+        <path
+          d="M23.9994 29.3277V44.5H19.9994V29.3289L14.4142 34.9141L11.5858 32.0857L19.7373 23.9342C20.9869 22.6845 23.0131 22.6845 24.2627 23.9342L32.4142 32.0857L29.5858 34.9141L23.9994 29.3277Z"
+          fill="#E1E1E1"
+        />
+      </svg>
+    </UploadIllustrationContainer>
+  );
+};
+
+export type FileUploadProps = {
+  /**
+   * an array of file types the user is able to upload. See https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept#unique_file_type_specifiers for examples
+   */
+  acceptedFileTypes?: string[];
+  /**
+   * If `true` the drag and drop area will not be rendered
+   */
+  isButtonOnly?: boolean;
+  /**
+   * The label for the `input` element.
+   */
+  label: string;
+  /**
+   * Function that is called when the list of ifles to upload is changed
+   */
+  onChange: (files: File[]) => void;
+  /**
+   * If `true` multiple files can be uploaded.
+   */
+  type?: (typeof fileUploadTypes)[number];
+} & Pick<
+  FieldComponentProps,
+  | "errorMessage"
+  | "hint"
+  | "HintLinkComponent"
+  | "id"
+  | "isDisabled"
+  | "isOptional"
+>;
+
+const FileUpload = ({
+  acceptedFileTypes,
+  errorMessage,
+  id,
+  isButtonOnly,
+  isDisabled = false,
+  isOptional,
+  hint,
+  HintLinkComponent,
+  label,
+  onChange,
+  type,
+}: FileUploadProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [filesToUpload, setFilesToUpload] = useState<File[]>([]);
+
+  useEffect(() => {
+    onChange(filesToUpload);
+  }, [filesToUpload, onChange]);
+
+  const handleFileInputChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const { files } = event.target;
+
+      if (files && files.length > 0) {
+        const mergedFiles = type ? [...filesToUpload, ...files] : [...files];
+
+        setFilesToUpload(mergedFiles);
+      }
+
+      // reset input value to allow re-upload of a file with the same name
+      event.target.value = "";
+    },
+    [type, filesToUpload]
+  );
+
+  const handleUploadButtonClick = useCallback(() => {
+    inputRef?.current?.click();
+  }, [inputRef]);
+
+  const handleFileRemoval = useCallback(
+    (name: string) => {
+      const deletedFileFilteredOut = filesToUpload.filter(
+        (file) => file.name != name
+      );
+      setFilesToUpload(deletedFileFilteredOut);
+    },
+    [filesToUpload]
+  );
+
+  const renderFileInput = useCallback(
+    ({ ariaDescribedBy, errorMessageElementId, id, labelElementId }) => {
+      const fileNames = filesToUpload.map((file) => file.name);
+
+      const Input = () => (
+        <input
+          accept={acceptedFileTypes?.toString()}
```

Comment:
`acceptedFileTypes` is already a string, no?

Reply:
Oh, I see what you're doing. I'd prefer to see this as `acceptedFileTypes?.join(",")` so that it's more clear what we're doing with this `string[]`